File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        count_neg = 0
        count_zero = 0

        for num in nums:
            if num < 0:
                count_neg -= 1
            elif num == 0:
                count_zero += 1

        if count_zero >= 1:
            return 0

        return 1 if count_neg % 2 == 0 else -1

File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int countNeg = 0;
        int countZero = 0;

        for (int num : nums) {
            if (num < 0) {
                countNeg--;
            } else if (num == 0) {
                countZero++;
            }
        }

        if (countZero >= 1) {
            return 0;
        }

        return countNeg % 2 == 0 ? 1 : -1;
    }
}